index.js

```diff
@@ -731,6 +748,19 @@ function getChoppedStreamHandler (client, onerror) {
       }
     }
 
+    // Provide a function on the client for it to signal this intake request
+    // to gracefully shutdown, i.e. finish up quickly.
+    client._intakeRequestGracefulExitFn = () => {
```

Comment:
Thinking out loud about the concurrency here -- if I'm reading this correctly there's a period of time between one request finishing and another request starting where `client._intakeRequestGracefulExitFn` will "belong" to the previous request.  If the `process` object emits the `beforeExit` event during this period our handler will still call `_intakeRequestGracefulExitFn`.  However, this shouldn't have any ill effect as the `completePart` function has sufficient guard clauses and will bail out if the request is already finished.

Reply:
Yes. Actually `intakeResTimer` should always be set null for that period. It might be more obvious to explicitly set this GracefulExitFn to null when concluding the request. Done in commit cab7b75